### DISCUSSION TITLE
Suggestion: add slot list to API documentation

### DISCUSSION THREAD

**Discussion by datvm** [material-components/material-web#5331] [Ideas]
### Description

Many elements like [List](https://material-web.dev/components/list/) or [Text field](https://material-web.dev/components/text-field/) has many slot names, sometimes with different names between components like `icon` vs `trailing-icon`. It would be nice to gather them all in one place like the `Properties` and `Methods` section of `API`.

Is it difficult to add? I assume you would need to add a JSDoc annotatation.

**Comment by asyncliz** [comment]
A welcome addition, but there isn't a jsdoc annotation to describe slot APIs, so we need to come up with our own and then implement it in our api analyzer scripts.

We can brainstorm suggestions here though!

**Comment by datvm** [comment]
Thanks for opening to the suggestion. Sorry I do not know how your analyzer script works or where it is. I think we can use something similar to `@fire`, like Tabs':

```
/**
 * @fires change {Event} Fired when the selected tab changes. The target's selected or
 * ...
 */
export class Tabs extends LitElement {
```

Maybe we can add something like this:

```
/**
 * @slots icon {Slot} Tabs may optionally show an icon.
 * Icons communicate the type of content within a tab.
 * Icons should be simple and recognizable.
 * ...
 */
export class Tab extends tabBaseClass {
```

Does this make sense? Is the analyzer script publicly available? I can try to take a look at it and see if I can help.

**Reply by asyncliz** [reply]
The scripts are located here https://github.com/material-components/material-web/tree/main/scripts/analyzer, and they use [@lit-labs/analyzer](https://github.com/lit/lit/tree/main/packages/labs/analyzer).

@e111077 might know more and have opinions.
